clang/lib/Sema/SemaOpenMP.cpp

```diff
@@ -23094,6 +23094,59 @@ void Sema::checkDeclIsAllowedInOpenMPTarget(Expr *E, Decl *D,
   checkDeclInTargetContext(E->getExprLoc(), E->getSourceRange(), *this, D);
 }
 
+class MyStmtVisitor final : public ConstStmtVisitor<MyStmtVisitor> {
+public:
+  SmallVector<Decl *> DeclVector;
+  Decl *TargetDecl;
+  void VisitDeclRefExpr(const DeclRefExpr *Node) {
+    Decl *DeclVar = nullptr;
+    if (const VarDecl *VD = dyn_cast<VarDecl>(Node->getDecl())) {
+      DeclVar = (Decl *)Node->getDecl();
+      DeclVar->addAttr((this->TargetDecl)->getAttr<OMPDeclareTargetDeclAttr>());
+      (this->DeclVector).push_back(DeclVar);
+    }
+  }
+  void VisitMyExpr(const Expr *Ex) {
+    for (Expr::const_child_iterator it = Ex->child_begin(); 
+         it != Ex->child_end(); ++it) {
+      if (isa<Expr>(*it))
+        VisitMyExpr(dyn_cast<Expr>(*it));
+      if (isa<DeclRefExpr>(*it))
+        Visit(*it);
```

Comment:
I am assuming this `Visit()` calls the `MyStmtVisitor::VisitDeclRefExpr()` indirectly via overriding, as `*it` is a `DeclRefExpr`. Why not call `MyStmtVisitor::VisitDeclRefExpr()` yourself, if you know that `*it` is a `DeclRefExpr` ?